### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by MechanicKim**
1. Updated sign in page
2. Added find password things

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
 import React, { Component } from 'react';
-import { Platform, StyleSheet, Alert } from 'react-native';
-import Login from './Login';
+import {
+  Alert,
+  Platform,
+  StyleSheet
+} from 'react-native';
+import SplashPage from './SplashPage';
```
Comment (ace0625) [inline src/components/App.js]: Sort

File: src/components/Commons/Hr.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    textAlign: 'center',
+  },
+  textContainer: {
+    height: 20,
+    marginLeft: 15,
+    marginRight: 15,
+  },
+});
+
+class Hr extends Component {
```
Comment (ace0625) [inline src/components/Commons/Hr.js]: Better change the file name.

File: src/components/Commons/Progress.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  View,
+} from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+
+const deviceWidth = Dimensions.get('window').width;
+
+// This is a progress line.
+// This component receives 'level' and 'step' to define line length
+class Progress extends Component {
+
```
Comment (ace0625) [inline src/components/Commons/Progress.js]: delete empty line

File: src/components/Login/FindPassStep1.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      email: '',
+    };
+  }
+
+  render() {
+    let onChangeText = (text) => this.onChangeText(text);
+    let requestSecretCode = () => this.requestSecretCode();
+
+    return (
+      <FindPassword title="Please input your email address"
```
Comment (ace0625) [inline src/components/Login/FindPassStep1.js]: organize the proptypes

File: src/components/Login/FindPassStep2.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      code: '',
+    };
+  }
+
+  render() {
+    let onChangeText = (text) => this.onChangeText(text);
+    let checkSecretCode = () => this.checkSecretCode();
+
+    return (
+      <FindPassword title="Please input your code"
```
Comment (ace0625) [inline src/components/Login/FindPassStep2.js]: organize proptypes

File: src/components/Login/FindPassStep3.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  render() {
+    let onChangePassword1 = (text) => this.onChangePassword1(text);
+    let onChangePassword2 = (text) => this.onChangePassword2(text);
+    let resetPassword = () => this.resetPassword();
+
+    return (
+      <FindPassword title="Reset your password"
+                    inputHint="New password"
```
Comment (ace0625) [inline src/components/Login/FindPassStep3.js]: organize proptypes

File: src/components/Login/FindPassStep3.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  onChangePassword1(text) {
+    this.state.password1 = text;
+  }
+
+  onChangePassword2(text) {
+    this.state.password2 = text;
+  }
+
+  resetPassword() {
+    if (this.state.password1 === '') {
```
Comment (ace0625) [inline src/components/Login/FindPassStep3.js]: Better use undefined instead of empty string

File: src/components/Login/FindPassword.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+ Alert,
+ StyleSheet,
+ Text,
+ TextInput,
+ TouchableWithoutFeedback,
+ View,
+} from 'react-native';
+
+class FindPassStep extends Component {
+
```
Comment (ace0625) [inline src/components/Login/FindPassword.js]: delete empty line

File: src/components/Login/FindPassword.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    return (
+
+      //  Render the screen on View.
+      <View style={styles.container}>
+        <View style={styles.titleContainer}>
+          <Text style={styles.title}>{this.props.title}</Text>
+        </View>
+        <View style={styles.inputContainer}>
+          <TextInput style={styles.input}
```
Comment (ace0625) [inline src/components/Login/FindPassword.js]: organize proptypes

File: src/components/Login/Login.js [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+import { Actions } from 'react-native-router-flux';
+import ErrorMeta from '../../utils/ErrorMeta';
+import LoginMeta from '../../utils/LoginMeta';
+import LinearGradient from 'react-native-linear-gradient';
+import LoginUtil from '../../utils/LoginUtil';
+import ServerUtil from '../../utils/ServerUtil';
+import styles from './Styles';
+
+class Login extends Component {
+
```
Comment (ace0625) [inline src/components/Login/Login.js]: delete empty line

File: src/components/Login/Login.js [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+    AsyncStorage.multiSet(
+       [['token', result.user.password], ['loginType', LoginMeta.LOGIN_TYPE_LOCAL]],
+       () => onSignInSuccess(result)
+    );
+  }
+
+  focusNextField(refNo) {
+    this.refs[refNo].focus();
+  }
+
```
Comment (ace0625) [inline src/components/Login/Login.js]: delete empty line

File: src/components/Login/SignUp.js [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+} from 'react-native';
+import { Actions } from 'react-native-router-flux';
+import ErrorMeta from '../../utils/ErrorMeta';
+import LinearGradient from 'react-native-linear-gradient';
+import LoginUtil from '../../utils/LoginUtil';
+import ServerUtil from '../../utils/ServerUtil';
+import styles from './Styles';
+
+class Login extends Component {
+
```
Comment (ace0625) [inline src/components/Login/SignUp.js]: delete empty line

File: src/components/Login/SignUp.js [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+        {/* Render facebook login button */}
+        <TouchableWithoutFeedback onPress={LoginUtil.signInWithFacebook}>
+          <View style={[styles.facebookLoginContainer, { marginTop: 43 }]}>
+            <Image style={styles.facebookLoginButton}
+                   source={require('../../resources/fb.png')} />
+            <Text style={styles.facebookLoginText}>Login with Facebook</Text>
+          </View>
+        </TouchableWithoutFeedback>
+
```
Comment (ace0625) [inline src/components/Login/SignUp.js]: delete empty line

File: src/components/Login/SignUp.js [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+        <View style={[styles.hrContainer, { marginTop: 10 }]}>
+          <View style={styles.hr}></View>
+          <View><Text style={styles.hrText}>or</Text></View>
+          <View style={styles.hr}></View>
+        </View>
+
```
Comment (ace0625) [inline src/components/Login/SignUp.js]: delete empty line

File: src/components/Login/SignUp.js [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+        <View style={styles.inputContainer}>
+          <TextInput style={styles.input}
+                     ref="1"
```
Comment (ace0625) [inline src/components/Login/SignUp.js]: organize proptypes

File: src/components/Login/SignUp.js (hunk 14 above)
Comment (MechanicKim) [reply src/components/Login/SignUp.js]: As you know, legibility is important factor, so I think empty line is necessary in some case. But, If we have to follow guideline, Okay! I'll

**Comment by lyw0149** [comment]
Thanks mingyu!! LGTM

**Comment by lyw0149** [comment]
LGTM

**Pull request merged**